### PR TITLE
MATM:enable retrieving original allowance

### PR DESCRIPTION
**Pull request by ross-rosario**
- `ManualApproval` struct now contains:
  - `allowance`, which is the original supplied (or modified) allowance.
  - `remainingAllowance` which keeps track of ongoing allowance.
- Added a few tests to MATM to ensure > 95% coverage.

### CODE REVIEW

**Comment by ross-rosario** [comment]
@pabloruiz55 @satyamakgec I renamed allowance variables according to your suggestion:
- `allowance` is back to keeping track of the remaining allowance.
- `approvedAllowance` stores the original allowance. I avoided calling it `original`Allowance because the word _original_ (to me) means the allowance value before the approval got _modified_ via `modifyManualApproval`, for example.

File: contracts/modules/TransferManager/MATM/ManualApprovalTransferManager.sol [hunk 1]
```diff
@@ -230,6 +231,7 @@ contract ManualApprovalTransferManager is ManualApprovalTransferManagerStorage,
                     allowance = allowance - _changeInAllowance;
                 }
             }
+            approval.approvedAllowance = allowance;
```
Comment (satyamakgec) [inline contracts/modules/TransferManager/MATM/ManualApprovalTransferManager.sol]: Sorry to re-iterate @remon-nashid but `initialAllowance` will be the better name for this

Comment (satyamakgec) [reply contracts/modules/TransferManager/MATM/ManualApprovalTransferManager.sol]: And one more thing, `approval.approvedAllowance` value should be changed as per the increase or decrease i.e
```
       if increase 
       approvedAllowance =  approvedAllowance + changeInAllowance
       if decrease
       approvedAllowance =  approvedAllowance - changeInAllowance
```

**Review by satyamakgec — CHANGES_REQUESTED** [review]

File: contracts/modules/TransferManager/MATM/ManualApprovalTransferManager.sol (hunk 1 above)
Comment (ross-rosario) [reply contracts/modules/TransferManager/MATM/ManualApprovalTransferManager.sol]: Done

**Review by satyamakgec — APPROVED** [review]

**Comment by ross-rosario** [comment]
@VictorVicente we're keeping this PR open so that you utilize `initialAllowace` field in the CLI. Basically, it's a new member variable we added to `ManualApproval` struct, in order to keep a record of the initial allowance, versus the ongoing allowance tracked by `allowance` member variable.

**Pull request merged**